iridauploader/parsers/base_parser.py

```diff
@@ -0,0 +1,34 @@
+class BaseParser:
+    """
+     This class is the abstract for the different parser objects.
+
+     Like the miseq and directory parser, a new parser class needs the following static methods
+         find_single_run(directory)
+         find_runs(directory)
+         get_required_file_list()
+         get_sample_sheet(directory)
+         get_sequencing_run(sample_sheet, ...)
```

Comment:
Yes and no.

Yes you can but because Python objects are mutable there is no way to really enforce an object to include all methods, other than throwing an excepting in the base class.

Also to do this we would extend the `abc` modules Abstract Base Class `ABC`, which comes with a bunch of additional things that need to be done to keep it pythonic. I have a branch that I have been working on now and again to switch all the parsers to use `ABC` but I felt that it is a bit out of scope for this issue, and is big enough to have it's own PR once it's done.